### ISSUE TITLE
`when` for activation is not ng2 conform

### ISSUE THREAD

**Issue by ribizli** [alexcastillo/ng2-notifications#4]
It's only my opinion, but triggering via setting a variable to true for one CD iteration is dangerous and non ng2 conform.
I recommended using a template variable to trigger a notification.

``` html
<systemNotification #notification ...></systemNotification>
<button (click)="notification.notify()">show</button>
```

But I think an even better way would be to use a service!

**Comment by alexcastillo** [comment]
Hi @ribizli,

I like your proposal.
I'll experiment with it and get back to you.

Thank you!

**Comment by alexcastillo** [comment]
@ribizli Please take a look and let me know if this is better.

**Comment by ribizli** [comment]
Definitely, but I still think that the best solution is to use a service. There is no sense to include any component into the view if it doesn't have any view role. Sorry if this wasn't clear in my original post and you did unnecessary work.

**Comment by alexcastillo** [comment]
I already created a ticket for making a service out of this. Please refer to #9 
But I still think a component is useful.

Thanks for the feedback!

**Comment by ribizli** [comment]
One more: I guess you changed to component (from directive) because to be able to use template variable. But you can do it with directive as well using `exportAs`.

**Comment by alexcastillo** [comment]
Yes, I was able to make it work with `exportAs` but didn't like how the syntax became more complex. 
Also, I changed it to a component so I could set the element to `display: none;`. That way it doesn't mess with the flow of the elements around it.

**Comment by ribizli** [comment]
Still I'm curious about your argument for "component is useful" ;)

**Comment by alexcastillo** [comment]
@ribizli I agree a service can satisfy all use cases.
I think it is important to let the developer choose how to use the future. Do you think we should only have it a service and remove the component?

**Comment by ribizli** [comment]
Yes, I see using component is more confusing for the users (developers) since it has nothing to do with the view. And if you think about usecases, a notification will come rarely from a view (by a user action), but from other service calls.

**Comment by alexcastillo** [comment]
You have a great point. Do you mind submitting a PR?

**Comment by ribizli** [comment]
I go now for holiday for two weeks. I can't promise anything 😉

**Comment by alexcastillo** [comment]
No problem, man. I'll have something by the time you come back.

Enjoy your holiday and thank you for the support!

**Comment by LenonLopez** [comment]
Looking to update this project to angular 4 (5 soon enough). I would also like to implement this notification as a service rather than a component. Been a while since this repo has been updated but I would still like to contribute here rather than start from scratch. @alexcastillo  thoughts?

update: As i'm doing the updates/ enhancements I realized that it is simply much easier to create a brand new project (angular 4.2.4) and port over the functionality into an injectable service. I realized after commenting that the version of angular used in this project (2.0.0-rc 1) is simply too outdated to try and manually make changes to.
